python/fbprophet/diagnostics.py

```diff
@@ -319,6 +319,55 @@ def rolling_mean_by_h(x, h, w, name):
     return pd.DataFrame({'horizon': res_h, name: res_x})
 
 
+def rolling_median_by_h(x, h, w, name):
+    """Compute a rolling median of x, after first aggregating by h.
+
+    Right-aligned. Computes a single median for each unique value of h. Each
+    median is over at least w samples.
+
+    For each h where there are fewer than w samples, we take samples from the previous h,
+    moving backwards. (In other words, we ~ assume that the x's are shuffled within each h.)
+
+    Parameters
+    ----------
+    x: Array.
+    h: Array of horizon for each value in x.
+    w: Integer window size (number of elements).
+    name: Name for metric in result dataframe
+
+    Returns
+    -------
+    Dataframe with columns horizon and name, the rolling median of x.
+    """
+    # Aggregate over h
+    df = pd.DataFrame({'x': x, 'h': h})
+    grouped = df.groupby('h')
+    df2 = grouped.size().reset_index().sort_values('h')
+    hs = df2['h'].values
+
+    res_h = []
+    res_x = []
+    # Start from the right and work backwards
+    i = len(hs) - 1
+    while i >= 0:
+        h_i = hs[i]
+        xs = grouped.get_group(h_i).x.tolist()
+        j = i - 1
+        while (len(xs) < w) and (j >= 0):
+            # Include points from the previous horizon. All of them if still
+            # less than w, otherwise just enough to get to w.
+            xs.append(x[j])
```

Comment:
Isn't `xs` a list, so this should be `xs.extend(x[j])`?

Reply:
`xs` is a list but `x[j]` is a scalar, so I believe `append` is correct.

Reply:
Sorry for the confusion, right after making that comment I realized that I had mistakenly thought x was something different than what it is, so what is here is correct. I wrote up a comment saying that my earlier comment was in error but I guess it didn't post for some reason. Everything looks good here, feel free to proceed!